src/xo-app/host/tab-network.js

```diff
@@ -128,81 +127,156 @@ class PifItem extends Component {
-        {_toggleDefaultLockingMode(
-          <Toggle
-            disabled={pifInUse}
-            onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
-            value={networks[pif.$network].defaultIsLocked}
-          />,
-          pifInUse && _('pifInUse')
-        )}
-      </td>
-      <td>
-        <StateButton
-          disabledLabel={_('pifDisconnected')}
-          disabledHandler={connectPif}
-          disabledTooltip={_('connectPif')}
-
-          enabledLabel={_('pifConnected')}
-          enabledHandler={disconnectPif}
-          enabledTooltip={_('disconnectPif')}
-
-          disabled={pif.attached && (pif.management || pif.disallowUnplug)}
-          handlerParam={pif}
-          state={pif.attached}
-        />
-        {' '}
-        <Tooltip content={pif.carrier ? _('pifPhysicallyConnected') : _('pifPhysicallyDisconnected')}>
-          <Icon
-            icon='network'
-            size='lg'
-            className={pif.carrier ? 'text-success' : 'text-muted'}
-          />
-        </Tooltip>
-      </td>
-      <td className='text-xs-right'>
-        <ActionRowButton
-          disabled={pif.physical || pif.disallowUnplug || pif.management}
-          handler={deletePif}
-          handlerParam={pif}
-          icon='delete'
-          tooltip={_('deletePif')}
-        />
-      </td>
-    </tr>
+      case 5:
+        return <pre>{pif.mac}</pre>
+      case 6:
+        return <div>{pif.mtu}</div>
+      case 7:
+        return <div className='text-xs-center'>
+          {_toggleDefaultLockingMode(
+            <Toggle
+              disabled={pifInUse}
+              onChange={() => editNetwork(pif.$network, { defaultIsLocked: !networks[pif.$network].defaultIsLocked })}
+              value={networks[pif.$network].defaultIsLocked}
+            />,
+            pifInUse && _('pifInUse')
+          )}
+        </div>
+      case 8:
+        return <div>
+          <StateButton
+            disabledLabel={_('pifDisconnected')}
+            disabledHandler={connectPif}
+            disabledTooltip={_('connectPif')}
+
+            enabledLabel={_('pifConnected')}
+            enabledHandler={disconnectPif}
+            enabledTooltip={_('disconnectPif')}
+
+            disabled={pif.attached && (pif.management || pif.disallowUnplug)}
+            handlerParam={pif}
+            state={pif.attached}
+                />
+          {' '}
+          <Tooltip content={pif.carrier ? _('pifPhysicallyConnected') : _('pifPhysicallyDisconnected')}>
+            <Icon
+              icon='network'
+              size='lg'
+              className={pif.carrier ? 'text-success' : 'text-muted'}
+                  />
+          </Tooltip>
+        </div>
+      default: return <div> Empty ! </div>
+    }
   }
 }
 
+const COLUMNS = [
+  {
+    itemRenderer: (pif, networks) => {
+      return <PifItem ID={0} key={pif.id} pif={pif} networks={networks} />
```

Comment:
Do not use the same component for all the columns. Either directly return the node you want it to be rendered as or create a smaller component that fits your needs.